### PR TITLE
feat(componets/headerbar): make all props optional

### PR DESCRIPTION
**Pull request by sdwvit**
**What is the problem this PR is trying to solve?**
For login page there is no need in brand link

**What is the chosen solution to this problem?**
so making it optional 

**Please check if the PR fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 1]
```diff
@@ -8,19 +8,19 @@ import Typeahead from '../Typeahead';
...
+function getComponents(Components) {
 	return Object.assign(
 		{
 			Action,
 			ActionDropdown,
 			ActionSplitDropdown,
 			Typeahead,
 		},
-		renderers || {},
+		Components || {},
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Please don't use upper case on arguments names.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 2]
```diff
@@ -231,20 +230,20 @@ function HeaderBar(props) {
 		User,
 		Help,
 		Products,
-	}, props.renderers || {});
+	}, getComponents(renderers) || {});
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: Would it be possible for `getComponents` to return the default value?

File: packages/components/src/HeaderBar/HeaderBar.component.js (hunk 1 above)
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: And please use default argument value instead of `||`.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 3]
```diff
@@ -216,13 +216,12 @@ function Products({ renderers, ...props }) {
...
 Products.propTypes = {
-	renderers: React.PropTypes.shape({
+	Components: React.PropTypes.shape({
 		ActionDropdown: React.PropTypes.func,
 	}),
 };
 
-function HeaderBar(props) {
-	const renderers = getRenderers(props.renderers);
+function HeaderBar({ logo, brand, env, search, help, user, products, renderers }) {
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: I don't see the benefit of spreading the argument here. It creates a ton of allocations. 
Please revert back to using the 'props'.

File: packages/components/src/HeaderBar/HeaderBar.component.js [hunk 4]
```diff
@@ -231,20 +230,20 @@ function HeaderBar(props) {
...
+				{ logo && <Components.Logo {...{ Components }} {...logo} />}
+				{ brand && <Components.Brand {...{ Components }} isSeparated={!!env} {...brand} />}
+				{ env && <Components.Environment {...{ Components }} {...env} /> }
 			</ul>
 			<ul className={classNames(theme['tc-header-bar-actions'], theme.right)}>
-				{ props.search && <Components.Search renderers={renderers} {...props.search} /> }
-				{ props.help && <Components.Help renderers={renderers} {...props.help} /> }
-				{ props.user && <Components.User renderers={renderers} {...props.user} /> }
-				{ props.products && <Components.Products renderers={renderers} {...props.products} /> }
+				{ search && <Components.Search {...{ Components }} {...search} /> }
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: I find the `{...{ Components }}` syntax confusing.
Do you think it could be improved.

File: packages/components/src/HeaderBar/__snapshots__/HeaderBar.snapshot.test.js.snap [hunk 5]
```diff
@@ -1526,7 +1526,7 @@ exports[`Storyshots HeaderBar default 1`] = `
                       other
                     </td>
                     <td>
-                      yes
+                      no
```
Comment (talend-glorieux) [inline packages/components/src/HeaderBar/__snapshots__/HeaderBar.snapshot.test.js.snap]: I don't get why the result was changed here.

File: packages/components/src/HeaderBar/HeaderBar.component.js (hunk 4 above)
Comment (sdwvit) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: yea, let me do some magic on it

File: packages/components/src/HeaderBar/HeaderBar.component.js (hunk 2 above)
Comment (sdwvit) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: sure

File: packages/components/src/HeaderBar/HeaderBar.component.js (hunk 3 above)
Comment (sdwvit) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: ok

File: packages/components/src/HeaderBar/HeaderBar.component.js (hunk 1 above)
Comment (sdwvit) [inline packages/components/src/HeaderBar/HeaderBar.component.js]: ok

**Comment by sdwvit** [comment]
@glorieux I have reverted all changes that are out of the scope of pr, please check

File: output/components.eslint.txt [hunk 6]
```diff
@@ -15,6 +15,9 @@ The react/require-extension rule is deprecated. Please use the import/extensions
   1:1  error  '@storybook/react' should be listed in the project's dependencies, not devDependencies  import/no-extraneous-dependencies
   4:2  error  Unexpected require()                                                                    global-require
 
+/home/travis/build/Talend/ui/packages/components/src/HeaderBar/HeaderBar.component.js
+  141:6  error  Expected indentation of 4 tab characters but found 5  react/jsx-indent
```
Comment (talend-glorieux) [inline output/components.eslint.txt]: Please fix that error.

Comment (jmfrancois) [inline output/components.eslint.txt]: looks weird this is not related to his change (which is line 241)

Comment (sdwvit) [inline output/components.eslint.txt]: @glorieux this is some defect from ci, it was fixed by last commit
ill trigger build again
